### ISSUE TITLE
Plugin Default Behavior Update

### ISSUE THREAD

**Issue by georgesnelling** [lyft/flyte#19]
{"json":\\{"exec_id":"***","node":"***","ns":"**-development","routine":"worker-13","src":"handler.go:216","tasktype":"spark","wf":"*****.SparkTasksWorkflow"}

,"level":"warning","msg":"No plugin found for Handler-type [spark], defaulting to [container]","ts":"2019-11-11T21:09:36Z"}

Defaulting Spark to container doesn't make sense and ideally we should fail cleanly at Propeller level  and expose it to users instead of executing it as a container task and leading to an unknown/weird container failures. I think this also applies to other tasks like Hive/Sidecar.

**Comment by kumare3** [comment]
I do not agree with this bug, the idea of task types is that we want to allow users define arbitrary client side task types (show them up in the UI) but the behavior of these tasks should be to actually run in a container. 
But, I do agree with one part, that we should have better messaging when we defaulted the task type. Maybe for task execution events, we can show the TaskType of the task and the ExecutionTaskType?

**Comment by kumare3** [comment]
@EngHabu, wdyt?

**Comment by EngHabu** [comment]
I agree the real bug is the seemingly transparent/implicit behavior of the system.
I like the idea of showing/exposing the plugin ID somewhere. 

We also, separately, talked about pinning plugins/plugin versions in the workflow closure and somehow deal with forced upgrades/ plugin changes... might be worth opening that discussion in the same veins...

**Comment by kumare3** [comment]
That is interesting, but should we close this issue and open a more appropriate issue?
